store/sqlstore/thread_store.go

```diff
@@ -18,6 +18,42 @@ import (
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
+// JoinedThread allows querying the Threads + Posts table in a single query, before looking up
+// users and unpacking into a model.ThreadResponse.
+type JoinedThread struct {
```

Comment:
Doesn't look like this struct and its method need to be exported.

Reply:
@vish9812, indeed we don't reference it explicitly, but as it's used as a target for `database/sql` (actually https://github.com/jmoiron/sqlx), the struct needs to be exported so that its fields can be reflected as part of querying the database.

You're right that we could keep `ToThreadResponse` within the package, though! Fixed -- thanks :)